server/src/domain/plugin_backend_handler.rs

```diff
@@ -0,0 +1,510 @@
+    LdapSearchResultEntry,
+};
+use lldap_auth::{login, registration, types::UserId};
+use lldap_domain::{
+    requests::{
+        CreateAttributeRequest, CreateGroupRequest, CreateUserRequest, UpdateGroupRequest,
+        UpdateUserRequest,
+    },
+    schema::Schema,
+    types::{AttributeName, Group, GroupDetails, GroupId, LdapObjectClass, User, UserAndGroups},
+};
+use lldap_domain_handlers::handler::{
+    BackendHandler, BindRequest, GroupBackendHandler, GroupListerBackendHandler,
+    GroupRequestFilter, LoginHandler, ReadSchemaBackendHandler, RequestContext,
+    SchemaBackendHandler, UserBackendHandler, UserListerBackendHandler, UserRequestFilter,
+    WithContextHandler,
+};
+use lldap_domain_model::error::Result;
+use lldap_plugin_engine::api::{
+    arguments::{ldap_bind_result::BindResult, ldap_search_result::SearchResult},
+    handler::{PluginHandler, PluginHandlerEvents},
+    types::PluginContext,
+};
+use lldap_plugin_kv_store::store::PluginKeyValueStore;
+
+use crate::{
+    domain::{
+        ldap::utils::LdapInfo, opaque_handler::OpaqueHandler, plugin::backend::ServerBackendAPI,
+        sql_backend_handler::SqlBackendHandler,
+    },
+    infra::{ldap_handler::InternalSearchResults, tcp_backend_handler::TcpBackendHandler},
+};
+
+use tracing::instrument;
+
+#[derive(Clone, Debug)]
+pub struct PluginBackendHandler {
+    pub(crate) backend_handler: SqlBackendHandler,
+    pub(crate) backend_api: &'static ServerBackendAPI<SqlBackendHandler>,
+    pub(crate) plugin_handler: PluginHandler<PluginKeyValueStore>,
+    pub(crate) request_context: RequestContext,
+}
+
+impl PluginBackendHandler {
+    pub fn new(
+        backend_handler: &SqlBackendHandler,
+        plugin_handler: PluginHandler<PluginKeyValueStore>,
+    ) -> Self {
+        let api: &'static ServerBackendAPI<SqlBackendHandler> =
+            Box::leak(Box::new(ServerBackendAPI {
+                backend_handler: backend_handler.clone(),
+                ldap_info: LdapInfo::from(
+                    backend_handler.config.ldap_base_dn.clone(),
+                    backend_handler.config.ignored_user_attributes.clone(),
+                    backend_handler.config.ignored_group_attributes.clone(),
+                ),
+            }));
+        PluginBackendHandler {
+            backend_handler: backend_handler.clone(),
+            backend_api: api,
+            plugin_handler: plugin_handler,
+            request_context: RequestContext::empty(),
+        }
+    }
+    pub fn new_plugin_context(&self) -> PluginContext<ServerBackendAPI<SqlBackendHandler>> {
+        PluginContext::new(
+            self.backend_api,
+            self.request_context.validation_results.clone(),
+        )
+    }
+    pub async fn initialize_plugins(&self) -> std::result::Result<(), String> {
+        let ctx = self.new_plugin_context();
+        self.plugin_handler.initialize_plugins(ctx).await
```

Comment:
That reminds me that in the final PR, I'll probably bicker about what gets printed when initializing plugins (name, version, permission/capabilities, update URL, whether there's a new version available, ...)